scripts/release/run_check_sync_criteria_job.py

```diff
@@ -39,6 +39,10 @@ def main(argv=sys.argv[1:]):
     add_argument_config_url(parser)
     add_argument_rosdistro_name(parser)
     add_argument_build_name(parser, 'release')
+    parser.add_argument(
+        'os_name', nargs='?',
```

Comment:
Adding an optional positional argument in the middle seems problematic to me?

Reply:
I thought so at first too, but it works as desired:
```python3
>>> import argparse
>>> parser = argparse.ArgumentParser()
>>> parser.add_argument('first')
>>> parser.add_argument('second', nargs='?')
>>> parser.add_argument('third')
>>> parser.parse_args(['1', '2', '3'])
Namespace(first='1', second='2', third='3')
>>> parser.parse_args(['1', '2'])
Namespace(first='1', second=None, third='2')
```

Reply:
I understand if you don't think it presents an intuitive user experience. My goal is just to provide a backwards-compatible change and make the argument required as soon as we reasonably can.

Reply:
The reason I'd like to put it in the middle is so it follows all the rest of the jobs, which list the args in the order `os_name` `os_code_name` `arch`.

Reply:
With a follow up PR shortly after converting it in a required argument :+1:

Reply:
Filed as #728. Thanks.